### ISSUE TITLE
Uploaded CSV data not appearing / not filling gaps in TSA

### ISSUE THREAD

**Issue by SRGDamia1** [ODM2/ODM2DataSharingPortal#430]
http://data.envirodiy.org/sites/4GBee_Bypass/ - the live data gapped here between 9/2 and 9/3.  I've uploaded the file which contains the missing data.  The upload gives a success message, but the gap persists.

**Comment by aufdenkampe** [comment]
@SRGDamia1, yes, we had a 3-4 service gap around Sep 2-3 for all sites.

The real issue is with uploading to fill the data gap.

@Volk3rJ recently had the same issue, but found that the data is actually getting uploaded to the database, because it is immediately available for download. However, the new data is not showing up in the Sparklines or TSA. This tells me that there's an issue with updating the InfluxDB database. There's a script that automatically does this, but it must not be triggering properly. 

I'll work with @htaolimno in the next week or so to figure out how to get this to work.

@SRGDamia1 & @ Volk3rJ, in the meanwhile, let me know of other things you notice.

**Comment by SRGDamia1** [comment]
The sparklines and TSA are definitely only being update on the half-hour instead of instantly.

But even half-an-hour later file uploaded gaps aren't filled.

**Comment by Volk3rJ** [comment]
I got different results with different stations. Some update the sparklines, some don't. The TSA seems to work now but it took a lot longer than 30 minutes for the data to be available there. Below are my 4 stations. All of them have data until 08/23/19.

* http://data.envirodiy.org/sites/DoEE_McF_PPF-09N/   (sparklines work for one sensor, outdated for the other)
* http://data.envirodiy.org/sites/DoEE_McF_PPF-09S/ (sparklines are outdated for all sensors)
* http://data.envirodiy.org/sites/DoEE_LAF_PPF-17/ (sparklines work for one sensor, outdated for the other)
* http://data.envirodiy.org/sites/DoEE_LAF_PPF-36/ (no sparklines at all)

**Comment by SRGDamia1** [comment]
Jacob Lemon was seeing gaps persisting at his stations - http://monitormywatershed.org/sites/ColdwaterMorse/

**Comment by SRGDamia1** [comment]
@Volk3rJ - can you tell me if the data is at least in the ODM2 database, even if it's not viewable on TSA?

**Comment by aufdenkampe** [comment]
See https://github.com/ODM2/ODM2DataSharingPortal/issues/443#issuecomment-570652082, which is a duplicate of this issue, for additional information.

**Comment by w3asa** [comment]
I reconfirmed the data gap problem when I was uploading data to [GMI_EC1](http://monitormywatershed.org/tsa/?sitecode=GMI_EC1&view=datasets&plot=false)
I have the µSD data chunked into 1 week files to the upload process doesn't choke (is that still an issue?).  I uploaded one file out of time sequence and that produced a gap in the TSA display.  The data is present in the download file however.

**Comment by aufdenkampe** [comment]
We have decided to replace TSA with a Time Series Visualization (TSV) page, starting with:
- #512

Check out the screenshots I just posted with https://github.com/ODM2/ODM2DataSharingPortal/issues/512#issuecomment-957935317. 
We think our new TSV addresses this issue.

Specifically, TSV fetches data directly from the main underlying ODM2 database, and does not require any of the translation layers (i.e., TSA Catalog or InfluxDB) that were causing previous troubles. Therefore, any data that gets successfully uploaded will definitely show up in the TSV plots. We can track any data upload bugs in separate issues.
